timemachine/cpp/src/kernels/k_neighborlist.cuh

```diff
@@ -201,11 +211,9 @@ void __global__ k_find_blocks_with_ixns(
 
     int neighborsInBuffer = 0;
 
-    const unsigned int atom_i_idx = blockIdx.x * blockDim.x + threadIdx.x;
-
-    RealType pos_i_x = atom_i_idx < NR ? row_coords[atom_i_idx * 3 + 0] : 0;
-    RealType pos_i_y = atom_i_idx < NR ? row_coords[atom_i_idx * 3 + 1] : 0;
-    RealType pos_i_z = atom_i_idx < NR ? row_coords[atom_i_idx * 3 + 2] : 0;
+    RealType pos_i_x = atom_i_idx < N ? coords[atom_i_idx * 3 + 0] : 0;
```

Comment:
this logic is making my head hurt a bit:

```
1) atom_i_idx = atom_i_idx < NR ? row_idxs[atom_i_idx] : N;
2) RealType pos_i_x = atom_i_idx < N ? coords[atom_i_idx * 3 + 0] : 0;
```
why not:

```
1) atom_i_idx = blockIdx.x * blockDim.x + threadIdx.x;
2) RealType pos_i_x = atom_i_idx < NR ? coords[atom_i_idx * 3 + 0] : 0;
```

i.e. is there a reason why we condition this over N?